packages/client/src/MessagePickupRepositoryClient.ts

```diff
@@ -75,13 +75,14 @@ export class MessagePickupRepositoryClient implements MessagePickupRepository {
    * @param {JsonRpcParamsMessage} data - The data received via the 'messageReceive' event.
    *
    * @param {string} data.connectionId - The ID of the connection associated with the message.
-   * @param {QueuedMessage[]} data.message - An array of queued messages received.
+   * @param {string} data.message - queued messages received.
```

Comment:
This is a limitation of the JSON-RPC library that needs to simple types such as strings? 

What about doing the `JSON.parse(data.message)` inside MessagePickupRepositoryClient's connect method, so the callback itself can have a cleaner typing.

I mean:

```
 client.addListener('messageReceive', (data) => {
          if (this.messageReceivedCallback) {
            this.messageReceivedCallback(JSON.parse(data) as MessageReceivedCallbackParams)
          } else {
            this.logger.log('Received message event, but no callback is registered:', data)
          }
        })
```

where `MessageReceivedCallbackParams` does include the QueuedMessage[] array and anything it may need.

Reply:
The problem originates from the message publication in addMessage, as Redis publish only accepts strings, so the message was being transmitted as string. The solution lies in performing a JSON.parse on the message as soon as it arrives at the subscribe in addLiveSession, which constructs the object that will be returned through the WebSocket in sendMessageToClientById.